### PR TITLE
[SCAL-81933, SCAL-73433]

### PR DESCRIPTION
**Pull request by sumit-thoughtspot**

### CODE REVIEW

**Review by thrinethra-thoughtspot — CHANGES_REQUESTED** [review]
Addressed the comments.

File: src/auth.ts [hunk 1]
```diff
@@ -17,13 +17,19 @@ const EndPoints = {
  * Check if we are logged into the ThoughtSpot cluster
  * @param thoughtSpotHost The ThoughtSpot cluster hostname or IP
  */
-async function isLoggedIn(thoughtSpotHost: string): Promise<boolean> {
+export async function getSessionInfo(
+    thoughtSpotHost: string,
+): Promise<{ loggedIn: boolean; data: any }> {
     const authVerificationUrl = `${thoughtSpotHost}${EndPoints.AUTH_VERIFICATION}`;
     const response = await fetch(authVerificationUrl, {
         credentials: 'include',
```
[the remark this answers is not in the document]
Comment (jbhanu-thoughtspot) [reply src/auth.ts:30]: @thrinethra-thoughtspot isLoggedIn will give sessionInfo only after logging in. So, we need handle the initial flow when it is not yet logged in as well.
1) Basic auth & Token auth: Use the response from the login call for storing sessionInfo.
2) Saml auth: After the login redirection, make a new getSessionInfo (or isLoggedIn) call.

[the remark this answers is not in the document]
Comment (thrinethra-thoughtspot) [reply src/auth.ts:26]: Resolved by calling initMixpanel in auth.ts init method and making call to fetch mixpanel token.

File: .idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (ashubham) [inline .idea/codeStyles/Project.xml:1]: remove these .idea files ?

**Review by ashubham — APPROVED** [review]

File: src/mixpanel-service.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        uploadMixpanelEvent(event.eventId, event.eventProps);
+    });
+}
+
+export async function initMixpanel(config: EmbedConfig): Promise<void> {
+    if (nodeEnv === TEST_ENV) {
+        return;
+    }
+    const { thoughtSpotHost } = config;
+    fetch(`${thoughtSpotHost}${EndPoints.CONFIG}`)
```
Comment (ashubham) [inline src/mixpanel-service.ts:55]: Please return this promise, so if someone wants to wait on this can do so.

**Review by ashubham — APPROVED** [review]

**Pull request merged**